Add get_node and get_leaf_nodes functions (#80)
ytree-project/ytree: doc/source/Arbor.rst

Each of these will return a generator of
:class:`~ytree.data_structures.tree_node.TreeNode` objects or field
values for all :class:`~ytree.data_structures.tree_node.TreeNode` objects

> in the tree, forest, or progenitor line. To get on specific node from a

Author: ```suggestion
in the tree, forest, or progenitor line. To get a specific node from a
```

tree, see :ref:`single-node-access`.

.. note:: Access by forest is supported even for datasets that do not

>    group trees by forest. If there is no requirement for the order in

Author: ```suggestion
   group trees by forest. If you have no requirement for the order in
```
